test/integration/arbitrum-mainnet/Loans.fork.t.sol

```diff
@@ -144,22 +151,223 @@ contract LoansForkTest is LoansTestBase {
+        skip(pair.durations[0] + maxGracePeriod / 2);
+
+        // Calculate expected partial late fee
+        (, uint lateFee) = pair.escrowNFT.currentOwed(loanBefore.escrowId);
+        assertGt(lateFee, 0);
+        assertLt(lateFee, underlyingAmount * lateFeeAPR * maxGracePeriod / (BIPS_BASE * 365 days));
+
+        closeAndCheckLoan(loanId, loanAmount, lateFee);
+
+        // Check escrow position's withdrawable (underlying + interest + partial late fee)
+        uint withdrawable = pair.escrowNFT.getEscrow(loanBefore.escrowId).withdrawable;
+        assertEq(withdrawable, underlyingAmount + interestFee + lateFee);
+
+        // Execute withdrawal and verify balance
+        vm.startPrank(escrowSupplier);
+        pair.escrowNFT.withdrawReleased(loanBefore.escrowId);
+        vm.stopPrank();
+
+        assertEq(
+            pair.underlying.balanceOf(escrowSupplier) - escrowSupplierBefore,
+            underlyingAmount + interestFee + lateFee
+        );
+    }
+
+    function testRollEscrowLoanBetweenSuppliers() public {
+        // Create first provider and escrow offers
+        (uint offerId1, uint escrowOfferId1) = _createEscrowOffers();
+        uint interestFee1 = pair.escrowNFT.interestFee(escrowOfferId1, underlyingAmount);
+
+        uint userUnderlyingBefore = pair.underlying.balanceOf(user);
+        (uint loanId, uint providerId, uint loanAmount) = _executeEscrowLoan(offerId1, escrowOfferId1);
+
+        // User has paid underlyingAmount + interestFee1 at this point
+        assertEq(userUnderlyingBefore - pair.underlying.balanceOf(user), underlyingAmount + interestFee1);
+
+        uint escrowSupplier1Before = pair.underlying.balanceOf(escrowSupplier);
+
+        // Create second escrow supplier
+        address escrowSupplier2 = makeAddr("escrowSupplier2");
+        deal(address(underlying), escrowSupplier2, bigUnderlyingAmount);
+
+        // Skip half the duration to test partial fees
+        skip(pair.durations[0] / 2);
+
+        // Get exact refund amount from contract
+        (, uint toLoans,) = pair.escrowNFT.previewRelease(loanId, 0);
+
+        // Create second escrow supplier's offer
+        vm.startPrank(escrowSupplier2);
+        pair.underlying.approve(address(pair.escrowNFT), underlyingAmount);
+        uint escrowOfferId2 = pair.escrowNFT.createOffer(
+            underlyingAmount,
+            pair.durations[0],
+            interestAPR,
+            maxGracePeriod,
+            lateFeeAPR,
+            0 // minEscrow
+        );
+        vm.stopPrank();
+
+        // Create roll offer using existing provider position
+        uint rollOfferId = createRollOffer(pair, provider, loanId, providerId, rollFee, rollDeltaFactor);
+
+        uint newEscrowFee = pair.escrowNFT.interestFee(escrowOfferId2, underlyingAmount);
+        uint userUnderlyingBeforeRoll = pair.underlying.balanceOf(user);
+
+        vm.startPrank(user);
+        pair.cashAsset.approve(address(pair.loansContract), loanAmount);
```

Comment:
no need to approve that much cash here. Should be just rollFee (if price hasn't moved)